File: code-coaches/medium/average-world-length/average_word_length.py
from math import ceil

# Short approach
# from string import punctuation


def get_average_word_length(words: list[str]) -> int:
    words_number = len(words)
    total_length = 0

    # Long approach
    for word in words:
        for char in word:
            if char.isalpha():
                total_length += 1

    # Short approach
    # total_length = sum(len(word.translate(str.maketrans('', '', punctuation))) for word in sentence)

    average_length = ceil(total_length / words_number)
    return average_length


def main():
    words = input().split(" ")
    print(get_average_word_length(words))


if __name__ == "__main__":
    main()

File: code-coaches/medium/average-world-length/average_word_length.rb
def get_average_word_length(words)
    words_number = words.length()
    total_length = 0

    words.each do |word|
        word.each_char do |char|
            if char.match(/^[[:alpha:]]$/)
                total_length += 1
            end
        end
    end

    average_length = (total_length.to_f / words_number.to_f).ceil
    return average_length
end


def main
    words = gets.chomp.split(" ")
    puts get_average_word_length(words)
end


if __FILE__ == $0
    main
end